proc/groupby.go

```diff
@@ -192,51 +203,54 @@ func (g *GroupBy) Pull() (zbuf.Batch, error) {
 	}
 }
 
-func (g *GroupByAggregator) createRow(keytype *zng.TypeRecord, ts nano.Ts, vals zcode.Bytes) *GroupByRow {
+func (g *GroupByAggregator) createRow(keyCols keyRow, ts nano.Ts, vals zcode.Bytes) *GroupByRow {
 	// Make a deep copy so the caller can reuse the underlying arrays.
 	v := make(zcode.Bytes, len(vals))
 	copy(v, vals)
 	return &GroupByRow{
-		keytype:  keytype,
+		keycols:  keyCols,
 		keyvals:  v,
 		ts:       ts,
 		reducers: compile.Row{Defs: g.reducerDefs},
 	}
 }
 
-func keysTypeRecord(zctx *resolver.Context, r *zng.Record, keys []GroupByKey) *zng.TypeRecord {
+func newKeyCols(kctx *resolver.Context, r *zng.Record, keys []GroupByKey) keyRow {
```

Comment:
Nit: It’s a little odd that this is called newKeyCols but it returns a keyRow. I’d expect the name to reflect the return type.